snuba/subscriptions/worker.py

```diff
@@ -96,6 +96,7 @@ def run_consistent() -> Result:
                     id=request.id,
                     body=copy.deepcopy(request.body),
                     query=copy.deepcopy(request.query),
+                    snql_anonymized=request.snql_anonymized,
```

Comment:
Same, I think we should be able to generate the formatted query from the original query right before this statement
https://github.com/getsentry/snuba/blob/master/snuba/web/query.py#L122-L130

Reply:
If we anonymize here, we won't be anonymizing the "original" version of the Query we receive. The parser itself runs a bunch of [post processors](https://github.com/getsentry/snuba/blob/d9c91a606f70c05e2d998e985a897d354fc97f93/snuba/query/snql/parser.py#L1361-L1381) on the Query which modify the AST, so we have to anonymize right after creating the initial AST. This just means it happens in `query/snql/parser.py::parse_snql_query()` and it has to be propagated through the Request.